framework/include/cppmicroservices/SharedLibrary.h

```diff
@@ -209,7 +209,7 @@ class US_Framework_EXPORT SharedLibrary
   bool IsLoaded() const;
 
 private:
-  ExplicitlySharedDataPointer<SharedLibraryPrivate> d;
+  std::unique_ptr<SharedLibraryPrivate> d;
```

Comment:
Based on @saschazelzer comment that ```ExplicitySharedDataPointer``` is equivalent to ```std::shared_ptr``` and that ```SharedLibrary``` is copy-able, I think this should be ```std::shared_ptr``` and not ```unique_ptr```. Creating a new ```unique_ptr``` from the pointer to an existing memory object already owned by a ```unique_ptr``` in the ```SharedLibrary``` copy ctor is not correct to me. This seems to violate the single ownership model of ```unique_ptr```.